terraform/java/ec2/default/main.tf

```diff
@@ -138,6 +138,19 @@ resource "null_resource" "main_service_setup" {
       # The application needs time to come up and reach a steady state, this should not take longer than 30 seconds
       sleep 30
 
+      # Check if the application is up. If it is not up, then exit 1.
+      attempt_counter=0
+      max_attempts=30
+      until $(curl --output /dev/null --silent --head --fail $(echo "http://localhost:8080" | tr -d '"')); do
+        if [ $attempt_counter -eq $max_attempts ];then
+          echo "Failed to connect to endpoint. "
+          exit 1
+        fi
+        printf '.'
+        attempt_counter=$(($attempt_counter+1))
+        sleep 10
+      done
```

Comment:
We have an opportunity to make this logging more meaningful instead of just printing a line of `.`